### DISCUSSION TITLE
Class Based Custom Decorator?

### DISCUSSION THREAD

**Discussion by tzarger** [MichalLytek/type-graphql#1230] [Q&A] [answered]
Is it possible to have a Class Based Decorator so it can also have injected services, same as AuthChecker? I searched and didn't see anyone asking this, or see any in the examples. I would like to lighten up the context creation as in many resolver methods, I hardly use some of the properties. I would like to create a parameter decorator for the ones that need a bit more.

**Comment by tzarger** [comment]
By the way for the question about Class Based Custom Decorator was for as you probably guessed, TSyringe DI reason. Sorry forgot to include that above.

**Comment by MichalLytek** [comment, marked as answer]
WDYM? You can use classes as middleware, while parameter decorator are intended to be fast and simple getter.

**Reply by tzarger** [reply]
I saw another conversation that said you had to wrap a class based middleware in a function to get the input parameters. 

https://github.com/MichalLytek/type-graphql/discussions/906#discussioncomment-821047

**Reply by MichalLytek** [reply]
That was an answer to that question. It's not your case.

You should use Extensions in order to put parameters for method and then read it from `info` in the generic middleware.

You can have a function that create a new class for every usage and expect it to work automagically with DI container.

**Reply by tzarger** [reply]
It is late here, sorry for my confusion. I thought decorators had to be registered with that container, but it seems like it works without.
